graph/registry.go

```diff
@@ -59,7 +60,12 @@ func NewV2Repository(repoInfo *registry.RepositoryInfo, endpoint registry.APIEnd
 		Transport: authTransport,
 		Timeout:   15 * time.Second,
 	}
-	endpointStr := endpoint.URL + "/v2/"
+	endpointStr := endpoint.URL
+	if strings.HasSuffix(endpointStr, "/") {
```

Comment:
perhaps just;
```go
endpointStr := strings.TrimRight(endpoint.URL, "/") + "/v2/"
```

To get rid of the if/else. It would also deal with cases where the trailing `/` is missing, but also if there are multiple trailing `///`